### ISSUE TITLE
Fine-tuning Grounded Conversation Generation (GCG) Task 

### ISSUE THREAD

**Issue by hungnh1125** [mbzuai-oryx/groundingLMM#52]
Thank for your amazing work. I would like to reproduce the results from the task of Grounded Conversation Generation. 
Could you please help me make the following information apparent? 
1. In your design (Figure 2 in the GLaMM study), did you file-tune just LMM or did you fine-tune on Region Encoder, LLM, and Pixel Decoder as well?
<img width="1133" alt="Screenshot 2024-06-06 at 10 14 53 AM" src="https://github.com/mbzuai-oryx/groundingLMM/assets/45062585/a9c74126-5567-4479-83ca-d09cb71cd8ea">
2. How much time do you spend fine-tuning?
I attempted to fine-tune, and 10 epochs require 50 hours to finish.

**Comment by mmaaz60** [comment]
Hi @hungnh1125,

I appreciate your interest in our work. Please note that during training, the global image encoder and grounding image encoder are kept frozen and the region encoder, projection layers (VL and L-P) and the pixel decoder are fully finetuned, while the LLM is LORA finetuned with `alpha = 8`.

The training instructions are provided in [this readme](https://github.com/mbzuai-oryx/groundingLMM/blob/main/docs/training.md#1-grounded-conversation-generation-gcg-%EF%B8%8F). Please note that  it took us around 20 hours to run GCG finetuning on  8 NVIDIA A100-40GB GPUs.

I hope it answers your questions. Good Luck, and let me know if you have any questions.

**Comment by hungnh1125** [comment]
@mmaaz60 Thank you so much for replying to me. 
Could you help me to explain why in your function [_create_seg_token_mask](https://github.com/mbzuai-oryx/groundingLMM/blob/main/model/GLaMM.py#L172), you add torch.zeros((mask.shape[0], 575)).bool().cuda(), and torch.zeros((mask.shape[0], 1)).bool().cuda()? 
575 and 1 are specific values, and I don't understand why you use these values. 

Thank you so much.

**Comment by mmaaz60** [comment]
Hi @hungnh1125,

Thank you for your interest in our work. This is because we are using image at 336x336 resolution with 14 patch size.

**Comment by hungnh1125** [comment]
@mmaaz60 I checked and saw that the size of the image embedding is [576, 4096]. Could you help me explain why you didn't add the torch? zeros((mask.shape[0], 576)).bool().cuda() before mask? What is the meaning of adding only torch.zeros((mask.shape[0], 575)).bool().cuda() before mask and torch.zeros((mask.shape[0], 1)).bool().cuda() after mask?
I want to extend your model to work with multiple images, so I need to understand the meaning of each parameter. 

Thank you so much.
